src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -123,6 +133,12 @@ function valueStride(ty: Type): number {
     );
   }
 
+  if (ty instanceof StructType) {
+    return ty.elementTypes.reduce((sum, cur) => {
+      return sum + valueStride(cur);
```

Comment:
This isn't taking alignment into consideration.

```
struct S {
  a : i32,
  b : vec4<f32>,
}
```

Would be misaligned by 12 bytes.